### PR TITLE
BlockListAppender: with custom appender, don't react to nested list settings changes

### PR DESCRIPTION
**Pull request by jsnajdr**
Unit tests written with React Testing Library don't like situations where:
- a state update is performed asynchronously, like in a `Promise.then` handler
- causes a rerender of a component (because the state value has changed)
- doesn't lead to any observable UI change (because the component didn't really use the changed state value and rendered the same thing as before)

In such a case the "update not wrapped in `act()`" warning is triggered -- something has changed asychronously and the test didn't catch it by waiting for the change to happen.

This situation happens when testing the mobile columns block. You create a new column block, synchronously perform the test actions on it (like changing vertical alignment) and then finish the test. However, there is another process going on: the column block initially has no `blockListSettings`, and `updateBlockListSettings` dispatch for it gets async scheduled. This dispatch affects rendering of the block inserter (the + icon in the mobile toolbar):

IMG_3904

Because this inserter asks whether it can insert the default block (`canInsertDefaultBlock`). Initially this is `false` because the column block doesn't yet have any list settings, and later becomes `true`. That leads to rerender of the inserter.

But when the inserter is rendered with custom UI (`renderAppender`) it doesn't really use the `canInsertDefaultBlock` value. Therefore, a state update that has no UI effect.

This PR fixes that by splitting `BlockListAppender` into multiple components. The outer checks the custom UI and returns immediately when custom UI is requested. Only when custom UI is not requested, does it render an inner component which does the `useSelect` with `canInsertDefaultBlock`.

### CODE REVIEW

**Review by tyxla — COMMENTED** [review]
The code is looking great, nice find Jarda!

Some testing instructions would be nice. I'm happy to defer testing to the more familiar eyes also.

**Comment by jsnajdr** [comment]
> Some testing instructions would be nice.

This is hard because you'd have to build the mobile app from this branch and test it. Create a "Columns" block with two columns, and then select one column, and verify that it has a working block inserter (the + icon) in the bottom toolbar:

IMG_3914

On the other hand this is exactly what the unit tests are testing, so you might want to trust them and give up manual testing 🙂

**Review by tyxla — APPROVED** [review]
This looks great, and tests well in web. Given tests pass, I don't see why it wouldn't work well in RN. I'd be happy to ship it 👍 

I'll have to devote half a day to set up my mobile build again, but for now I'll pass on giving this a spin on mobile. 

I'll leave it to you @jsnajdr as to whether to wait for an approval from mobile folks or ship immediately.

**Pull request merged**
